guides/keras_nlp/fnet_classification.py

```diff
@@ -0,0 +1,380 @@
+
+"""
+Let's also define our parameters/hyperparameters.
+"""
+BATCH_SIZE = 64
+EPOCHS = 3
+MAX_SEQUENCE_LENGTH = 40
+VOCAB_SIZE = 15000
+
+EMBED_DIM = 128
+INTERMEDIATE_DIM = 512
+
+"""
+## Loading the Dataset
+Tensorflow Datasets (TFDS) is a library that provides a unified API for working
+with datasets that are stored in the TensorFlow format. We will use TFDS to load
+the SST-2 dataset.
+"""
+train_ds, val_ds, test_ds = tfds.load(
+    "huggingface:sst", split=["train", "validation", "test"], shuffle_files=True
+)
+
+"""
+## Preprocessing the Dataset
+We need to perform these two steps:
+1. Convert the input sentences to lowercase.
+2. The label is a sentiment score (float) lying in the range [0, 1]. We will
+convert it to either 0 or 1, keeping 0.5 as the threshold. This is a necessary
+step because the task we want to solve is a classification task.
+"""
+
+
+def preprocess_dataset(dataset):
+    dataset = dataset.map(
+        lambda x: {
+            "sentence": tf.strings.lower(x["sentence"]),
+            "label": tf.cast(x["label"] >= 0.5, dtype=tf.int32),
+        }
+    )
+    return dataset
+
+
+train_ds = preprocess_dataset(train_ds)
+val_ds = preprocess_dataset(val_ds)
+test_ds = preprocess_dataset(test_ds)
+
+
+"""
+Let's analyse the train-validation-test split. We'll also print a few samples.
+"""
+print("Number of Training Examples: ", len(train_ds))
+print("Number of Validation Examples: ", len(val_ds))
+print("Number of Test Examples: ", len(test_ds))
+
+for element in train_ds.take(5):
+    print(element)
+
+
+"""
+### Tokenizing the Data
+We'll be using `WordPieceTokenizer` from KerasNLP to tokenize the text.
+`WordPieceTokenizer` takes a WordPiece vocabulary and has functions for
+tokenizing the text, and detokenizing sequences of tokens.
+
+
+Before we define the tokenizer, we first need to train it on the dataset
+we have. WordPiece Tokenizer is a subword tokenizer; training it on a corpus gives
+us a vocabulary of subwords. A subword tokenizer is a compromise between word tokenizers
+(word tokenizers have the issue of many OOV tokens), and character tokenizers
+(characters don't really encode meaning like words do). Luckily, TensorFlow Text makes it very
+simple to train WordPiece on a corpus. Reference: https://www.tensorflow.org/text/guide/subwords_tokenizer
+
+For more details about WordPiece, please visit [this
+blog](https://ai.googleblog.com/2021/12/a-fast-wordpiece-tokenization-system.html).
+
+Note: The official implementation of FNet uses the SentencePiece Tokenizer.
+"""
+
+
+def train_word_piece(text_samples, vocab_size, reserved_tokens):
+    bert_tokenizer_params = dict(lower_case=True)
+
+    bert_vocab_args = dict(
+        # The target vocabulary size
+        vocab_size=vocab_size,
+        # Reserved tokens that must be included in the vocabulary
+        reserved_tokens=reserved_tokens,
+        # Arguments for `text.BertTokenizer`
```

Comment:
Same comments as other guide... `bert_tokenizer_params={"lower_case"=True}` remove `Arguments for `text.BertTokenizer` comment, remove `learn_params`.